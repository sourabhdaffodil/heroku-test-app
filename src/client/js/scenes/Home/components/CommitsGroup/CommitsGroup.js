/* eslint no-underscore-dangle: 0 */
import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { media } from '../../../../theme/style-utils';
import Commit from '../Commit/Commit';

const Ul = styled.ul`
  margin: 10px 0;
  padding: 0;
  border: 1px solid ${props => props.theme.color('background')};
  margin-left: 20px;
  
  ${media.phone`
    margin: 0;
    border-right: 0;
    border-left: 0;
  `}
`;

const renderCommits = commits => (
  commits.map(commit => (
    <Commit commit={commit} key={commit._id} />
  ))
);

const CommitGroup = ({ commits }) => (<Ul>{renderCommits(commits)}</Ul>);

CommitGroup.propTypes = {
  commits: PropTypes.array.isRequired,
};

export default CommitGroup;
