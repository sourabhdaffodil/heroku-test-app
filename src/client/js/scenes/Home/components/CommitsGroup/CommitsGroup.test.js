import React from 'react';
import { shallow } from 'enzyme';
import CommitsGroup from './CommitsGroup';

describe('Commits group component', () => {
  it('Must render an empty list', () => {
    const wrapper = shallow(<CommitsGroup commits={[]} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Must render list of commits', () => {
    const commits = [
      {
        _id: '59d678d9f51fea02ea1209b7',
        date: new Date('2016-06-04'),
        title: 'A title',
        description: 'Description...',
        image: 'https://avatars0.githubusercontent.com/u/4216535?v=4&s=72',
        hash: 'd9094385acf7d47c7c5c6f7ceb6866b54f9b602d',
        hashAlias: 'd909438',
      },
    ];
    const wrapper = shallow(<CommitsGroup commits={commits} />);
    expect(wrapper).toMatchSnapshot();
  });
});
