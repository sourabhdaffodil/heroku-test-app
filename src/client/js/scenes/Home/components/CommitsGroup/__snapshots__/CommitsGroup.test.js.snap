// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Commits group component Must render an empty list 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommitGroup
    commits={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [],
    },
    "ref": null,
    "rendered": Array [],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Commits group component Must render list of commits 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommitGroup
    commits={
      Array [
        Object {
          "_id": "59d678d9f51fea02ea1209b7",
          "date": 2016-06-04T00:00:00.000Z,
          "description": "Description...",
          "hash": "d9094385acf7d47c7c5c6f7ceb6866b54f9b602d",
          "hashAlias": "d909438",
          "image": "https://avatars0.githubusercontent.com/u/4216535?v=4&s=72",
          "title": "A title",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Commit
          commit={
            Object {
              "_id": "59d678d9f51fea02ea1209b7",
              "date": 2016-06-04T00:00:00.000Z,
              "description": "Description...",
              "hash": "d9094385acf7d47c7c5c6f7ceb6866b54f9b602d",
              "hashAlias": "d909438",
              "image": "https://avatars0.githubusercontent.com/u/4216535?v=4&s=72",
              "title": "A title",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "59d678d9f51fea02ea1209b7",
        "nodeType": "class",
        "props": Object {
          "commit": Object {
            "_id": "59d678d9f51fea02ea1209b7",
            "date": 2016-06-04T00:00:00.000Z,
            "description": "Description...",
            "hash": "d9094385acf7d47c7c5c6f7ceb6866b54f9b602d",
            "hashAlias": "d909438",
            "image": "https://avatars0.githubusercontent.com/u/4216535?v=4&s=72",
            "title": "A title",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Commit
            commit={
              Object {
                "_id": "59d678d9f51fea02ea1209b7",
                "date": 2016-06-04T00:00:00.000Z,
                "description": "Description...",
                "hash": "d9094385acf7d47c7c5c6f7ceb6866b54f9b602d",
                "hashAlias": "d909438",
                "image": "https://avatars0.githubusercontent.com/u/4216535?v=4&s=72",
                "title": "A title",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "59d678d9f51fea02ea1209b7",
          "nodeType": "class",
          "props": Object {
            "commit": Object {
              "_id": "59d678d9f51fea02ea1209b7",
              "date": 2016-06-04T00:00:00.000Z,
              "description": "Description...",
              "hash": "d9094385acf7d47c7c5c6f7ceb6866b54f9b602d",
              "hashAlias": "d909438",
              "image": "https://avatars0.githubusercontent.com/u/4216535?v=4&s=72",
              "title": "A title",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
